test/MOIWrapper.jl

```diff
@@ -0,0 +1,195 @@
+    MOIT.lintest(solver, linconfig, ["lin3","lin4"])
+
+    solver_nopresolve = GurobiOptimizer(OutputFlag=0, InfUnbdInfo=1)
+    MOIT.lintest(solver_nopresolve, linconfig)
+end
+
+@testset "Integer Linear tests" begin
+    intconfig = MOIT.TestConfig()
+    solver = GurobiOptimizer(OutputFlag=0)
+    MOIT.intlineartest(solver, intconfig, ["int3"])
+
+    # 3 is ranged
+end
+@testset "ModelLike tests" begin
+    solver = GurobiOptimizer()
+    MOIT.nametest(solver)
+    @testset "validtest" begin
+        MOIT.validtest(solver)
+    end
+    @testset "emptytest" begin
+        MOIT.emptytest(solver)
+    end
+    @testset "orderedindicestest" begin
+        MOIT.orderedindicestest(solver)
+    end
+    @testset "canaddconstrainttest" begin
+        MOIT.canaddconstrainttest(solver, Float64, Complex{Float64})
+    end
+    @testset "copytest" begin
+        solver2 = GurobiOptimizer()
+        MOIT.copytest(solver,solver2)
+    end
+end
+
+@testset "Gurobi Callback" begin
+    @testset "Generic callback" begin
+        m = GurobiOptimizer(OutputFlag=0)
+        x = MOI.addvariable!(m)
+        MOI.addconstraint!(m, MOI.SingleVariable(x), MOI.GreaterThan(1.0))
+        MOI.set!(m, MOI.ObjectiveFunction{MOI.ScalarAffineFunction{Float64}}(),
+            MOI.ScalarAffineFunction{Float64}(
+                [MOI.ScalarAffineTerm{Float64}(1.0, x)],
+                0.0
+            )
+        )
+
+        cb_calls = Int32[]
+        function callback_function(cb_data::Gurobi.CallbackData, cb_where::Int32)
+            push!(cb_calls, cb_where)
+            nothing
+        end
+
+        MOI.set!(m, Gurobi.CallbackFunction(), callback_function)
+        MOI.optimize!(m)
+
+        @test length(cb_calls) > 0
+        @test Gurobi.CB_MESSAGE in cb_calls
+        @test Gurobi.CB_PRESOLVE in cb_calls
+        @test !(Gurobi.CB_MIPSOL in cb_calls)
+    end
+
+    @testset "Lazy cut" begin
+        m = GurobiOptimizer(OutputFlag=0, Cuts=0, Presolve=0, Heuristics=0, LazyConstraints=1)
+        MOI.Utilities.loadfromstring!(m,"""
+            variables: x, y
+            maxobjective: y
+            c1: x in Integer()
+            c2: y in Integer()
+            c3: x in Interval(0.0, 2.0)
+            c4: y in Interval(0.0, 2.0)
+        """)
+        x = MOI.get(m, MOI.VariableIndex, "x")
+        y = MOI.get(m, MOI.VariableIndex, "y")
+
+        # We now define our callback function that takes two arguments:
+        #   (1) the callback handle; and
+        #   (2) the location from where the callback was called.
+        # Note that we can access m, x, and y because this function is defined
+        # inside the same scope
+        cb_calls = Int32[]
+        function callback_function(cb_data::Gurobi.CallbackData, cb_where::Int32)
+            push!(cb_calls, cb_where)
+            if cb_where == Gurobi.CB_MIPSOL
+                sol = [0.0, 0.0]
+                Gurobi.cbget_mipsol_sol(cb_data, cb_where, sol)
+                x_val, y_val = sol
```

Comment:
The only thing is it's not clear how to inject the MIP solution back into MOI-land.